web-portlet/src/java/org/sakaiproject/portlets/PortletIFrame.java

```diff
@@ -636,7 +637,14 @@ else if (SPECIAL_WORKSITE.equals(special))
 					    String infoUrl = StringUtils.trimToNull(s.getInfoUrl());
 					    if (infoUrl != null)
 					    {
-						    context.put("info_url", FormattedText.escapeHtmlFormattedTextarea(infoUrl));
+                                                   //Check if infoUrl is relative? and append the server url
+                                                   String serverUrl = ServerConfigurationService.getServerUrl();
+                                                   String serverName = new URL(serverUrl).getHost();
+                                                   Pattern checkPattern = Pattern.compile(String.format("^(https?:)?//%s:?\\d*/", serverName));
+                                                   if(!checkPattern.matcher(infoUrl).find()){
```

Comment:
But we already have existing values in the database these might not have been filtered.